### PR TITLE
docs: update deploy workflow

### PR DESCRIPTION
**Pull request by lily-de**
**What was wrong**

1. The goose home page was not pretty because of some symlink issues between the README and docs/index.md
2. The deploy_docs workflow was borked / not installing deps

**Solution**

1. Removed the symlink between the readme and the homepage file (docs/index.md) and just copied over the full readme into index.md and updated some asset paths as needed. Also removed video because I could not get it to play on the home page (it still works in the readme)
2. Moved to using uv to manage doc deployment

thanks @lamchau @lukealvoeiro and @damienrj for working with me on this!

### CODE REVIEW

**Comment by lamchau** [comment]
@lily-de should we close out #120 in favor of this? looks like warnings/errors don't show up in ci anymore

File: .github/workflows/deploy_docs.yaml [hunk 1]
```diff
@@ -4,37 +4,25 @@ on:
...
     - name: Checkout repository
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
 
-    - name: Set up Python
-      uses: actions/setup-python@v4
-      with:
-        python-version: '3.x'
+    - name: Install UV
+      run: curl -LsSf https://astral.sh/uv/install.sh | sh
```
Comment (lamchau) [inline .github/workflows/deploy_docs.yaml]: :nit: how about using the official workflow?

```yaml
 uses: astral-sh/setup-uv@v3
```

File: README.md [hunk 2]
```diff
@@ -26,11 +26,12 @@ Goose is your on-machine developer agent, automating engineering tasks seamlessl
 
 <p align="center">
 <a href="#unique-features-of-goose-compared-to-other-ai-assistants">Unique features</a> 🤖 •
-<a href="#what-block-employees-have-to-say-about-goose"> Block Employees on Goose</a> <img src=docs/assets/logo.png height="15", width="15" alt="Block Emoji"/> •
+<a href="#what-block-employees-have-to-say-about-goose"> Block Employees on Goose</a> <img src=assets/logo.png height="15", width="15" alt="Block Emoji"/> •
```
Comment (lilydelalande) [inline README.md]: Pro: renders in the docs now because we are using the path `assets/logo.png`
Con: readme is now not pretty

File: README.md [hunk 3]
```diff
@@ -180,14 +181,12 @@ Let us know what you think in our [Discussions][discussions] or the [**`#goose`*
 [configuration]: https://block-open-source.github.io/goose/configuration.html
 [cli]: https://block-open-source.github.io/goose/plugins/cli.html
 [providers]: https://block-open-source.github.io/goose/providers.html
-[use-cases]: https://block-open-source.github.io/goose/guidance/applications.html
+[use-cases]: https://block-open-source.github.io/goose/guidance/applications
```
Comment (lilydelalande) [inline README.md]:
```suggestion
[use-cases]: https://block-open-source.github.io/goose/guidance/applications.html
```

File: docs/scripts/gen_ref_pages.py [hunk 4]
```diff
@@ -14,6 +20,10 @@
 utils_modules = []
 
 for path in sorted(src.rglob("*.py")):
+    logging.info("Processing file: {}".format(path))
+    module_path = path.relative_to(src).with_suffix("")
+    if module_path.parts[-1] == "diff":  # Specifically check for the 'diff' case
+        logging.info("Diff path found: {}".format(path))
```
Comment (lilydelalande) [inline docs/scripts/gen_ref_pages.py]:
```suggestion
    module_path = path.relative_to(src).with_suffix("")
```

File: docs/scripts/gen_ref_pages.py [hunk 5]
```diff
@@ -2,6 +2,12 @@
 
 import mkdocs_gen_files
 
+import logging
+
+# Set up logging
+logging.basicConfig(level=logging.INFO, filename='mkdocs_gen_debug.log', filemode='w', format='%(name)s - %(levelname)s - %(message)s')
+
+
```
Comment (lilydelalande) [inline docs/scripts/gen_ref_pages.py]:
```suggestion

```

File: docs/scripts/gen_ref_pages.py [hunk 6]
```diff
@@ -14,6 +14,7 @@
 utils_modules = []
 
 for path in sorted(src.rglob("*.py")):
+    module_path = path.relative_to(src).with_suffix("")
```
Comment (lilydelalande) [inline docs/scripts/gen_ref_pages.py]:
```suggestion
```

**Review by baxen — APPROVED** [review]
LGTM! 

Can be a separate PR but i think in the future we should try to use https://facelessuser.github.io/pymdown-extensions/extensions/snippets/ 
to include the readme literal so we don't risk them growing out of sync

**Comment by lilydelalande** [comment]
Added a follow up ticket to better sync the readme and docs homepage: [linear](https://linear.app/squareup/issue/AIT-90/use-the-snippets-extension-from-pymdown-to-keep-the-readme-and-docs) (for internal tracking); [github issue](https://github.com/block-open-source/goose/issues/128)

**Pull request merged**